pkg/render/tiers/tiers.go

```diff
@@ -0,0 +1,333 @@
+	render.KibanaNamespace,
+	render.ManagerNamespace,
+	common.TigeraPrometheusNamespace,
+	"tigera-skraper",
+	render.ECKOperatorNamespace,
+	render.PacketCaptureNamespace,
+	rmeta.APIServerNamespace(operatorv1.TigeraSecureEnterprise),
+	common.CalicoNamespace,
+)
+
+var defaultTierOrder = 100.0
+
+func Tiers(cfg *Config) render.Component {
+	return tiersComponent{cfg: cfg}
+}
+
+type Config struct {
+	Openshift      bool
+	ManagedCluster bool
+}
+
+type tiersComponent struct {
+	cfg *Config
+}
+
+func (t tiersComponent) ResolveImages(is *operatorv1.ImageSet) error {
+	return nil
+}
+
+func (t tiersComponent) Objects() ([]client.Object, []client.Object) {
+	objsToCreate := []client.Object{
+		t.allowTigeraTier(),
+		t.allowTigeraClusterDNSPolicy(),
+		t.allowTigeraAPIServerPolicy(),
+		t.allowTigeraKubeControllersPolicy(),
+		t.allowTigeraPacketCapturePolicy(),
+	}
+
+	// Delete equivalent policies under different namespaced names previously managed outside the operator.
+	objsToDelete := []client.Object{}
+	if !t.cfg.Openshift {
+		objsToDelete = append(objsToDelete, &v3.NetworkPolicy{ObjectMeta: metav1.ObjectMeta{Name: "allow-tigera.kube-dns", Namespace: "kube-system"}})
+		objsToDelete = append(objsToDelete, &v3.NetworkPolicy{ObjectMeta: metav1.ObjectMeta{Name: "allow-tigera.kube-dns-egress", Namespace: "kube-system"}})
+	}
+
+	return objsToCreate, objsToDelete
+}
+
+func (t tiersComponent) Ready() bool {
+	return true
+}
+
+func (t tiersComponent) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeAny
+}
+
+func (t tiersComponent) allowTigeraTier() *v3.Tier {
+	return &v3.Tier{
+		TypeMeta: metav1.TypeMeta{Kind: "Tier", APIVersion: "projectcalico.org/v3"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: networkpolicy.TigeraComponentTierName,
+			Labels: map[string]string{
+				"projectcalico.org/system-tier": "true",
+			},
+		},
+		Spec: v3.TierSpec{
+			Order: &defaultTierOrder,
+		},
+	}
+}
+
+func (t tiersComponent) allowTigeraClusterDNSPolicy() *v3.NetworkPolicy {
+	var dnsPolicySelector string
+	var dnsPolicyNamespace string
+	if t.cfg.Openshift {
+		dnsPolicySelector = "dns.operator.openshift.io/daemonset-dns == 'default'"
+		dnsPolicyNamespace = "openshift-dns"
+	} else {
+		dnsPolicySelector = "k8s-app == 'kube-dns'"
+		dnsPolicyNamespace = "kube-system"
+	}
+
+	return &v3.NetworkPolicy{
+		TypeMeta: metav1.TypeMeta{Kind: "NetworkPolicy", APIVersion: "projectcalico.org/v3"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ClusterDNSPolicyName,
+			Namespace: dnsPolicyNamespace,
+		},
+		Spec: v3.NetworkPolicySpec{
+			Order:    &networkpolicy.HighPrecedenceOrder,
+			Tier:     networkpolicy.TigeraComponentTierName,
+			Selector: dnsPolicySelector,
+			Ingress: []v3.Rule{
+				{
+					Action: v3.Allow,
+					Source: v3.EntityRule{
+						NamespaceSelector: "all()",
+						Selector:          DNSIngressNamespaceSelector,
+					},
+				},
+				{
+					Action: v3.Pass,
+				},
+			},
+			Egress: []v3.Rule{
```

Comment:
The original CC policies had the egress split out separately so that users could restrict it if required. Since we are moving to a model where users do not edit the allow-tigera tier, I've combined them into one policy. The user can create policy in adjacent tiers to override

Reply:
@caseydavenport Looking at this again, I'm wondering if we should rethink inheriting the egress rule into the operator... seems a bit too broad of scope for the operator/allow-tigera. WDYT?

Reply:
Ah, sorry I think this was a misunderstanding of policy application on my part. I didn't consider the fact that if we restrict the allowed egress in this tier, that (even with the Pass) all other egress will be denied unless explicitly allowed in a later tier. 

Do let me know if I understand that correctly.

Will resolve in current PR (by ensuring the current rule encapsulates all known NS and removing the skraper namespace)

Reply:
Ah, yeah I think I just missed the "Pass" statement. 

I'd probably still leave it as-is and keep to our stance of translating 1:1 for the first pass, and then making changes in a subsequent iteratino.